ultravox/training/train.py

```diff
@@ -282,48 +297,103 @@ def main() -> None:
             # fsdp_transformer_layer_cls_to_wrap='LlamaDecoderLayer',
         ),
     )
-    if args.do_train:
-        # Training loop
-        logging.info("Starting training...")
-        t_start = datetime.now()
-        logging.info(f"train start time: {t_start}")
-        if args.val_steps:
-            trainer.evaluate()
-        trainer.train()
-        trainer.save_model(args.output_dir)
-        t_end = datetime.now()
-        logging.info(f"train end time: {t_end}")
-        logging.info(f"elapsed: {t_end - t_start}")
-
-    if args.do_eval:
-        logging.info("Starting evaluation...")
-        t_start = datetime.now()
-        logging.info(f"eval start time: {t_start}")
-
-        # Merge LoRA weights for better inference performance.
-        # Note: this is irreversible and changes model saving format
-        model.merge_and_unload()
-        inference = infer.LocalInference(
-            model=model,
-            processor=processor,
-            tokenizer=text_tokenizer,
-            device=args.device,
-            dtype=dtype,
+
+    # Training loop
+    logging.info("Starting training...")
+    t_start = datetime.now()
+    logging.info(f"train start time: {t_start}")
+    if args.val_steps:
+        trainer.evaluate()
+    trainer.train()
+    if trainer.is_world_process_zero():
```

Comment:
is this different from `if is_master:`

Reply:
In one node no. In multiple nodes, it enforce both that local_rank is zero and that node_rank is also zero, hence the word "world" process zero.

I can revert to using `is_master` for now.

Reply:
Understood. Does it require updates to the usage of `is_master` (e.g., eval) in other parts of the code when operating in multiple nodes?

Reply:
Then maybe use `is_master` for now.